python/ql/lib/semmle/python/ApiGraphs.qll

```diff
@@ -485,6 +485,40 @@ module API {
       )
     }
 
+    /**
+     * Holds if `result` is the result of awaiting `awaitedValue`.
+     */
+    cached
+    DataFlow::Node awaited(DataFlow::Node awaitedValue) {
+      // `await` x
+      // - `awaitedValue` is `x`
+      // - `result` is `await x`
+      exists(Await await |
+        result.asExpr() = await and
+        await.getValue() = awaitedValue.asExpr()
+      )
+      or
+      // `async for x in l`
+      // - `awaitedValue` is `l`
+      // - `result` is `l` (should perhaps be `x`, but that should really be a read)
+      exists(AsyncFor asyncFor |
+        result.asExpr() = asyncFor.getTarget() and
+        // Morally, we should perhaps use asyncFor.getIter() = awaitedValue.asExpr(),
+        // but that is actually behind a read step rather than a flow step.
+        asyncFor.getTarget() = awaitedValue.asExpr()
+      )
+      or
+      // `async with x as y`
+      // - `awaitedValue` is `x`
+      // - `result` is `x` (should probably be `y` but it might not exist)
+      exists(AsyncWith asyncWith |
+        result.asExpr() = asyncWith.getContextExpr() and
+        // Morally, we should perhaps use asyncWith.getOptionalVars() = awaitedValue.asExpr(),
+        // but that might not exist.
+        asyncWith.getContextExpr() = awaitedValue.asExpr()
+      )
```

Comment:
This seems to be able to handle the following example, so I'm assuming there is a default "use" step that adds flow from `x -> y`.

```py
    async with asyncpg.create_pool() as pool:
        await pool.execute("sql")  # $ getSql="sql"
```

In your PR, you've used 
```
API::moduleImport("asyncpg").getMember("create_pool").getReturn().getAwaited()
```
 
and I'm guessing that `API::moduleImport("asyncpg").getMember("create_pool").getReturn()` would also be able to find the `pool` in `pool.execute(...)` above. The thing this code piece does is allow you to consistently write `.getAwaited()` when using API graphs for things that should be awaited, because there now is an `Label::await()` edge from `x -> x`.

If we wanted to ensure that people _always_ write `.getAwaited()` when using API graphs, I think the above snippet should be changed so we get a ``Label::await()`` edge `x -> y` instead (if `y` is not there, we don't need an edge anyway, so that's fine), and then we need to _remove_ the general "use" edge from `x -> y` on `async with`. Without having thought too deeply about this, I think this sound like a solid approach, but maybe we should confer with @tausbn about it as well.